### DISCUSSION TITLE
Easily mix Dynamic and Static widgets/facets

### DISCUSSION THREAD

**Discussion by zergormagn** [algolia/instantsearch#5151] [General]
Is it possible to have a couple static widgets+facets.  A few hidden facets, and everything with a certain facet name as DynamicWidgets?

Currently dynamicwidgets has these parameters:   facets=[] or facets=['* ']   
I would want to do facets=['_*'] for example.
So that all facets starting with an underscore in the name get displayed as dynamic refinementLists.

**Comment by Haroenv** [comment]
The facets parameter is about which facets to request, and is set to * to be able to read the subsequent query from cache (as the parameters don't change).

What you're looking for is probably `transformItems` where you can filter out all facets that don't start with `_` https://www.algolia.com/doc/api-reference/widgets/dynamic-facets/js/#widget-param-transformitems

**Reply by zergormagn** [reply]
That makes sense. But would there be a way to optimize the amount of data that is retrieved at a time?  For instance, I can have more then 100 dynamic facets for few thousand hits. Is there a way to only retrieve the top 10 and provide a show more button to get more of the facets on request?

**Reply by Haroenv** [reply]
Unless you know in advance what those facets are, unfortunately not, but I've forwarded this to the relevant backend team so they're aware of the feature request.
